script/package-electron-builder.ts

```diff
@@ -9,8 +9,8 @@ const globPromise = promisify(glob)
 
 import { getDistPath, getDistRoot } from './dist-info'
 
-function getArchitecture() {
-  switch (process.arch) {
+function getArchitecture(targetArch?: string) {
+  switch (targetArch) {
```

Comment:
this should probably be changed to be more like `getDistArchitecture`. there is no need to set `TARGET_ARCH` and `npm_config_arch` when they both convey the same information. `TARGET_ARCH` is a variable internal to the dugite-native CI buildscripts. nothing is expecting that here.

basically these variables ( `npm_config_arch` and `npm_config_target_arch` ) are names recognized by node-gyp which is used to build native node modules (and used by electron)  as well as other node programs. technically both should be set for highest compatibility
https://www.electronjs.org/docs/latest/tutorial/using-native-node-modules#using-npm https://github.com/nodejs/node-gyp

ie:
```js
function getArchitecture() {
  // If a specific npm_config_arch is set, we use that one instead of the OS arch (to support cross compilation)
  if (process.env.npm_config_arch === 'arm64' ) {
    return '--arm64'
  }
  if (process.env.npm_config_arch === 'x64' ) {
    return '--x64'
  } 
  if ( process.env.npm_config_arch === 'armv7l' ) {
    return '--armv7l'
  }
  // If no specific npm_config_arch is set, use current process.arch
  switch (process.arch) {
    case 'arm64':
      return '--arm64'
    case 'arm':
      return '--armv7l'
    default:
      return '--x64'
  }
}
```

Reply:
although tenically you don't need to set `npm_config_arch` or `npm_config_target_arch` at all if you just call `electron-builder` with the appropriate --arm64, --armv7l, --x64 (which is what this script does) since electron-builder sets these variables for you based on the --arm64, --armv7l, --x64 input
https://github.com/electron-userland/electron-builder/blob/c1cc2e9c24784d05e14ca292adc2452e9c0237f6/packages/app-builder-lib/src/util/yarn.ts#L43-L48

the problem is I think (from remembering your scripts from memory) that you build with yarn directly and only use electron-builder to package the application basically. so you would need to set `npm_config_arch` or `npm_config_target_arch` yourself for any native packages that yarn (which invokes npm) builds.